### ISSUE TITLE
Connectors page only showing "psi-probe-web"

### ISSUE THREAD

**Issue by e1d** [psi-probe/psi-probe#687]
I downloaded and deployed the latest snapshot (Version 3.0.0-SNAPSHOT).

Everything works fine, except on the Connectors page, it doesn't show for example "ajp-bio-8009" connector, or "http-nio-8080". It only showed "psi-probe-web".

**Comment by hazendaz** [comment]
Can you post a screen shot?  Also can you try milestone 3 and let me know if the same problem exists there?  Currently milestone 3 is ahead of snapshot.  Thanks.


Sent from Outlook Mobile






On Fri, Mar 4, 2016 at 12:17 AM -0800, "e1d" <notifications@github.com> wrote:





I downloaded and deployed the latest snapshot (Version 3.0.0-SNAPSHOT).

Everything works fine, except on the Connectors page, it doesn't show for example "ajp-bio-8009" connector, or "http-nio-8080". It only showed "psi-probe-web".

---
Reply to this email directly or view it on GitHub:
https://github.com/psi-probe/psi-probe/issues/687

**Comment by hazendaz** [comment]
Also what version of tomcat are you using?


Sent from Outlook Mobile






On Fri, Mar 4, 2016 at 12:17 AM -0800, "e1d" <notifications@github.com> wrote:





I downloaded and deployed the latest snapshot (Version 3.0.0-SNAPSHOT).

Everything works fine, except on the Connectors page, it doesn't show for example "ajp-bio-8009" connector, or "http-nio-8080". It only showed "psi-probe-web".

---
Reply to this email directly or view it on GitHub:
https://github.com/psi-probe/psi-probe/issues/687

**Comment by e1d** [comment]
I'm using tomcat 8 & 7. Both returning same result.
Here is my screenshot 
![connector display](https://cloud.githubusercontent.com/assets/10696061/13558784/9097e770-e445-11e5-88c0-7df5ab441871.jpg)

**Comment by e1d** [comment]
Also, I got this error when creating war file with maven. What could be the problem?

`[INFO]
[INFO] --- git-commit-id-plugin:2.1.13:revision (git-commit-id) @ core ---
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] PSI Probe .......................................... SUCCESS [ 27.791 s]
[INFO] PSI Probe Core ..................................... FAILURE [04:55 min]
[INFO] PSI Probe Tomcat 5.0.x Adaptor ..................... SKIPPED
[INFO] PSI Probe Tomcat 5.5.x Adaptor ..................... SKIPPED
[INFO] PSI Probe Tomcat 6.0.x Adaptor ..................... SKIPPED
[INFO] PSI Probe Tomcat 7.0.x Adaptor ..................... SKIPPED
[INFO] PSI Probe Tomcat 8.0.x Adaptor ..................... SKIPPED
[INFO] PSI Probe Web Application .......................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 05:23 min
[INFO] Finished at: 2016-03-07T08:42:20+08:00
[INFO] Final Memory: 18M/95M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal pl.project13.maven:git-commit-id-plugin:2.1.13:revision (git-commit-id) on project core: .git directory could not be found! Please spec         ify a valid [dotGitDirectory] in your pom.xml -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :core
`

**Comment by hazendaz** [comment]
I looked at a older cut of the software from late last year.  Seems the issue has been in there for a while.  I'll keep this issue open until we can figure it out.  In a production copy I have, it listed the older naming style rather than as you see it now.

As to the build issue, it would appear that you downloaded a zip and tried to run from that.  The project needs cloned in order to run the build.  Per the error, the .git directory doesn't exist.  If that is how you pulled it, you can either perform a clone so the .git shows up or create a repo under the parent folder of the project so you end up with a .git file.  The build needs it.

**Comment by e1d** [comment]
Your production copy is in what version?
How about the milestone3? Does it displayed correctly?
Sorry guys, I cannot build the war since I cannot clone the git correctly. I think it is beacause our firewall.

**Comment by hazendaz** [comment]
My production copy is well after 2.4.0 was released and sometime before the milestones I believe.  I'll have to setup an environment to see if 2.4.0 was ok.  I suspect this will be easy to fix as it shows everything just doesn't label correctly.


Sent from Outlook Mobile






On Sun, Mar 6, 2016 at 6:56 PM -0800, "e1d" <notifications@github.com> wrote:





Your production copy is in what version?
How about the milestone3? Does it displayed correctly?
Sorry guys, I cannot build the war since I cannot clone the git correctly. I think it is beacause our firewall.

---

**Comment by e1d** [comment]
From what I see, it is not just labelling error, the connector traffic is not updating/showed.

**Comment by hazendaz** [comment]
Thanks for finding these issues.  If you are able and find the root cause please submit a pull request.  As soon as the issues are flushed out I'll cut another milestone.  This issue clearly now started before milestones.  I first noticed grids were not working sometime shortly after July last year.  I'm pretty sure it occurred on bump to java 6 and all that came with that.  I'll start looking deeper tonight.


Sent from Outlook Mobile






On Sun, Mar 6, 2016 at 9:03 PM -0800, "e1d" <notifications@github.com> wrote:





From what I see, it is not just labelling error, the connector traffic is not updating/showed.

---

**Comment by hazendaz** [comment]
This issue is present in the 2.4.0 release as well.  As that was released on 7/28, it's close to what I've been marking on each 3.0.0 milestone as a defect.  So starting point to find it should be roughly around that area and before.

**Comment by e1d** [comment]
I tried copying over the connector.jsp from 2.3.x version to 3.x.x version. And it worked fine.
So what should I do? I'm sorry because I'm a newbie with github.

**Comment by hazendaz** [comment]
You copied it over without any changes?  I think I tracked down where it occured but it's not jumping out at me.  Pull request #527 (commit 1046ea9) seems like it introduced the issue.  I wouldn't expect copying it over would fix the issue though as that was not part of that change.  That change was just moving around resource files.  I'm looking now to see if I can figure out the issue there.  If I run the commmit just prior to that it builds and works as expected.  If I run that commit, it fails to work as expected.

**Comment by e1d** [comment]
Yes, I copied it over without doing any other changes.

I think you can see the differences between the connector.jsp between the 2.3.x version and the later version. The differences was minor.

**Comment by hazendaz** [comment]
ok - I guess you copied it over without doing so within the build itself.  The issue was maven related.  Maven was taking ${name} and replacing that with the project name.  The commit I called out changed where resources were being filtered which directly resulted in everything being filtered and thus exposing this issue.  The quick fix will be to simply not filter this specific page as it does not require it.  I think I'm going to take that approach as I will need to backport that to 2.4.x as well since it was broken as well.  I really appreciate all the info you provided.  This resolves long standing issue and one I called out on each milestone.  It will be tomorrow night before I get this into the code base.

In the meantime, your way works for a quick workaround.  Or you can go into the web.xml and do this.

```xml
						<resource>
							<directory>${project.basedir}/src/main/webapp</directory>
							<filtering>true</filtering>
                                                </resource>
```

```xml
						<resource>
							<directory>${project.basedir}/src/main/webapp</directory>
							<filtering>true</filtering>
							<!-- Do not filter the connectors page as it contains ${name} -->
							<excludes>
								<excludes>**/*connectors.jsp</excludes>
							</excludes>
						</resource>
						<resource>
							<directory>${project.basedir}/src/main/webapp</directory>
							<filtering>false</filtering>
							<includes>
								<includes>**/*connectors.jsp</includes>
							</includes>
```

**Comment by MALfunction84** [comment]
Man, I love how putting "Fixes #687" in the merge commit closed this issue. /s
